crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -48,109 +82,197 @@ def all_platforms
         registry.keys
       end
 
+      def repositories(platform)
+        registry[platform]["repos"].keys
+      end
+
       def check_all_repos
         repochecks = []
         all_platforms.each do |platform|
           repositories(platform).each do |repo|
-            check = self.new(platform, repo)
-            if platform != Crowbar::Product.ses_platform && Crowbar::Product.is_ses?
-              #FIXME skip the repo check until SES is ported to SLE12-SP1
-              check.registry["required"] = "optional"
-            end
-            repochecks << check
+            repochecks << self.new(platform, repo)
           end
         end
         repochecks
       end
 
-      def admin_platform
-        NodeObject.admin_node.target_platform
+      def provided_and_enabled?(feature, platform = nil)
+        answer = false
+
+        if platform.nil?
+          all_platforms.each do |p|
+            if provided_and_enabled?(feature, p)
+              answer = true
+              break
+            end
+          end
+        else
+          found = false
+          answer = true
+
+          repositories(platform).each do |repo|
+            provided_features = registry[platform]["repos"][repo]["features"] || []
+
+            next unless provided_features.include? feature
+            found = true
+
+            r = self.new(platform, repo)
+            answer &&= r.active?
+          end
+
+          answer = false unless found
+        end
+
+        answer
       end
 
-      def web_port
-        Proposal.where(barclamp: "provisioner").first.raw_attributes["web_port"]
+      def platform_available?(platform)
+        available = true
+
+        repositories(platform).each do |repo|
+          r = self.new(platform, repo)
+          if r.required == "mandatory"
```

Comment:
Style/Next: Use `next` to skip iteration. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)